src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -35,6 +35,16 @@ namespace NUnit.Engine.Services
     /// </summary>
     public class DefaultTestRunnerFactory : InProcessTestRunnerFactory, ITestRunnerFactory
     {
+        private IProjectService _projectService;
+
+        public override void StartService()
+        {
+            _projectService = ServiceContext.GetService<IProjectService>();
+            Status = _projectService != null && ((IService)_projectService).Status == ServiceStatus.Started
```

Comment:
Sorry, previous comment is incorrect. The individual interfaces need to be separate from IService because some of them are public and IService is not / should not be available outside the engine. We are stuck with some sort of cast one way or the other. Casting to the type we asked for is guaranteed to be safe because that's how ServiceManager works. Casting to IService is guaranteed to be safe because only an IService can be added to ServiceManager in the first place.